### DISCUSSION TITLE
Could I get the enriched motif of peaks to a specific gene (The results of 'p2g')

### DISCUSSION THREAD

**Discussion by ChangliangWang** [GreenleafLab/ArchR#707] [Questions / Documentation] [answered]
Dear ArchR developers,

I am wondering could I use  'peakAnnoEnrichment' function (which needs a SummarizedExperiment object returned by markerFeatures() ) to get the enriched motifs of the peaks that link to a specific gene.	

Thanks very much for your help and sorry for any troubles to you.

Best regards,
Changliang.

**Comment by rcorces** [comment]
This might be possible but I'm not sure how the statistics would work. Wouldnt this only be for a handful of peaks? Seems difficult to identify enrichment from such a small number of observations?

**Reply by ChangliangWang** [reply]
Dear Ryan Corces,

Thanks very much for your quick reply. The number of peaks linking to my interested gene is very small, which is really a problem for me. I will think twice about how to figure my problem. Thanks very much for your help.

Have a nice day.

Best regards,
Changliang.

**Comment by NoemieL** [comment, marked as answer]
Hi. I am also interested on looking at special gene peaks and see with motif is enrich in this gene. For this I did a subset of markersPeaks but when I did peakAnnoEnrichment I get the error that Peaks from seMarker do not match peakSet in ArchRProj! I tested also just removing one line of the markersPeaks and give the same error. Could it be possible to authorize the seMArker to be different from the peakSet? Or do you suggest to do it differently.

This is what I did:
MM_OnlyPre_peak <- loadArchRProject(path = "Save-MM_OnlyPre_peak")

table(MM_OnlyPre_peak$Translocation)

markersPeaks <- getMarkerFeatures(
  ArchRProj = MM_OnlyPre_peak, 
  useMatrix = "PeakMatrix", 
  groupBy = "Translocation",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

markersPeaks

df=rowData(markersPeaks)
peakset=as.data.frame(getPeakSet(MM_OnlyPre_peak),row.names=NULL)

markersPeaks2=markersPeaks[row.names(df[which(df$seqnames=="chr18" & df$start %in% peakset[which(peakset$nearestGene=="BCL2"), "start"]),]),]

MM_OnlyPre_peak <- addMotifAnnotations(ArchRProj = MM_OnlyPre_peak, motifSet = "cisbp", name = "Motif", force=TRUE)

enrichMotifs <- peakAnnoEnrichment(
  seMarker = markersPeaks2,
  ArchRProj = MM_OnlyPre_peak,
  peakAnnotation = "Motif",
  cutOff = "FDR <= 0.05 & Log2FC >= 1"
)
Error in peakAnnoEnrichment(seMarker = markersPeaks2, ArchRProj = MM_OnlyPre_peak,  : 
  Peaks from seMarker do not match peakSet in ArchRProj!

Thanks for your help

**Reply by rcorces** [reply]
As in my previous reply, you will not have statistical power to determine enrichment. If what you really mean is you want to know which motifs are present in those peaks, then just use the `MotifMatrix`

**Reply by NoemieL** [reply]
I am sorry but I don't understand how I can get the motif find in a specific gene with MotifMatrix. Could you please be more explicit of what you are thinking? Thanks

**Reply by rcorces** [reply]
@NoemieL - my apologies for the typo. You want to use `getPeakAnnotation()` to get a binary representation of which peaks have which motifs:

```
> projHeme5 <- addMotifAnnotations(ArchRProj = projHeme5, motifSet = "cisbp", name = "Motif")
No methods found in package ‘IRanges’ for request: ‘score’ when loading ‘TFBSTools’
ArchR logging to : ArchRLogs/ArchR-addMotifAnnotations-3971645dd6259b-Date-2021-11-30_Time-18-06-14.log
If there is an issue, please report to github with logFile!
2021-11-30 18:06:15 : Gettting Motif Set, Species : Homo sapiens, 0.001 mins elapsed.
Using version 2 motifs!
2021-11-30 18:06:16 : Finding Motif Positions with motifmatchr!, 0.023 mins elapsed.
2021-11-30 18:08:39 : Creating Motif Overlap Matrix, 2.408 mins elapsed.
2021-11-30 18:08:41 : Finished Getting Motif Info!, 2.443 mins elapsed.
ArchR logging successful to : ArchRLogs/ArchR-addMotifAnnotations-3971645dd6259b-Date-2021-11-30_Time-18-06-14.log
> anno <- getPeakAnnotation(ArchRProj = projHeme5, name = "Motif")
> anno
$Name
[1] "Motif"

$motifs
PWMatrixList of length 870
names(870): TFAP2B_1 TFAP2D_2 TFAP2C_3 TFAP2E_4 TFAP2A_5 ... SMAD5_866 SMAD9_867 SOX6_868 TBX18_869 TBX22_870

$motifSummary
DataFrame with 870 rows and 3 columns
                 name                                   ID      strand
          <character>                          <character> <character>
TFAP2B_1       TFAP2B    ENSG00000008196_LINE2_TFAP2B_D_N1           *
TFAP2D_2       TFAP2D       ENSG00000008197_LINE6_TFAP2D_D           *
TFAP2C_3       TFAP2C    ENSG00000087510_LINE7_TFAP2C_D_N3           *
TFAP2E_4       TFAP2E      ENSG00000116819_LINE19_TFAP2E_I           *
TFAP2A_5       TFAP2A   ENSG00000137203_LINE29_TFAP2A_D_N3           *
...               ...                                  ...         ...
SMAD5_866       SMAD5 ENSG00000113658_LINE19362_SMAD5_I_N1           *
SMAD9_867       SMAD9 ENSG00000120693_LINE19375_SMAD9_I_N1           *
SOX6_868         SOX6  ENSG00000110693_LINE19574_SOX6_I_N7           *
TBX18_869       TBX18 ENSG00000112837_LINE19949_TBX18_I_N1           *
TBX22_870       TBX22 ENSG00000122145_LINE20002_TBX22_I_N1           *

$Positions
[1] "/corces/home/rcorces/ArchR/tutorial_201127/Save-ProjHeme5/Annotations/Motif-Positions-In-Peaks.rds"

$Matches
[1] "/corces/home/rcorces/ArchR/tutorial_201127/Save-ProjHeme5/Annotations/Motif-Matches-In-Peaks.rds"
```
What you are looking for is stored in the file indicated by `$Matches` as a sparse matrix (peaks by motifs)

**Reply by NoemieL** [reply]
Thank you! It is exactly what I needed.

**Reply by lzh-fugwaa** [reply]
Hi, I am also encounter this error, Could you tell me how to sovle it?
